### PR TITLE
CPLAT-17301 Update AriaPropsMixin

### PR DESCRIPTION
**Pull request by sydneesampson-wk**
## Motivation
Update AriaPropsMixin with missing attributes

## Changes
Added missing Aria attributes
Added missing Aria roles
Updated outdated W3 links
Add new documentation

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

File: lib/src/component/aria_mixin.dart [hunk 1]
```diff
@@ -21,6 +21,8 @@ import 'dart:collection';
 import 'package:over_react/over_react.dart' show PropDescriptor, PropsMeta, ConsumedProps;
 import 'package:over_react/src/component_declaration/annotations.dart';
 
+import '../../over_react.dart';
+
```
Comment (brianphillips-wk) [inline lib/src/component/aria_mixin.dart:25]: You can remove this

```suggestion
```

File: lib/src/component/aria_mixin.dart [hunk 2]
```diff
@@ -152,6 +164,13 @@ abstract class _$AriaPropsMixin {
   @Accessor(key: 'aria-describedby')
   dynamic describedby;
 
+   Accessor(key: 'aria-description')
```
Comment (brianphillips-wk) [inline lib/src/component/aria_mixin.dart:167]:
```suggestion
   @Accessor(key: 'aria-description')
```

File: lib/src/component/aria_mixin.dart [hunk 3]
```diff
@@ -594,6 +623,21 @@ abstract class _$AriaPropsMixin {
...
+
+@Accessor(key: 'aria-rowindex') 
+dynamic rowindex;
+
+@Accessor(key: 'aria-rowspan') 
+dynamic rowspan;
+
+
+
+
```
Comment (brianphillips-wk) [inline lib/src/component/aria_mixin.dart:640]: Be sure to run `ddev format` to get all the spacing lined up

File: lib/src/component/aria_mixin.dart [hunk 4]
```diff
@@ -594,6 +699,40 @@ abstract class _$AriaPropsMixin {
...
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-rowindex>
+  @Accessor(key: 'aria-rowindex')
+  int rowindex;
+
+  /// Defines a text alternative of aria-rowindex
+  ///
+  /// The aria-rowindextext should only be included in addition to, not as a replacement of, the aria-rowindex
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-rowindextext>
+  @Accessor(key: 'aria-rowindextext')
+  String aria-rowindextext
```
Comment (brianphillips-wk) [inline lib/src/component/aria_mixin.dart:728]: Quick fix:

```suggestion
  String rowindextext;
```

File: lib/src/component/aria_mixin.dart [hunk 5]
```diff
@@ -1640,9 +1848,12 @@ class Role {
   /// technologies through the platform accessibility API that the currently focused tab is selected.
   ///
   ///
-  /// See: <http://www.w3.org/TR/wai-aria/roles#tab>
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role>
   static const String tab = 'tab';
 
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/table_role>
+  static const String table = 'table'
```
Comment (brianphillips-wk) [inline lib/src/component/aria_mixin.dart:1855]:
```suggestion
  static const String table = 'table';
```

**Review by brianphillips-wk — DISMISSED** [review]
Slight tweaks here—Dart requires semicolons, hence the build failures

File: lib/src/component/aria_mixin.dart [hunk 6]
```diff
@@ -41,6 +42,8 @@ abstract class $AriaPropsMixin {
 abstract class _$AriaPropsMixin {
   Map get props;
 
+  ///See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes>
```
Comment (greglittlefield-wf) [inline lib/src/component/aria_mixin.dart:45]: This comment will get attached to `activedescendant`; could we move it to the doc comment above the `_ $AriaPropsMixin` class?

File: lib/src/component/aria_mixin.dart [hunk 7]
```diff
@@ -179,10 +252,22 @@ abstract class _$AriaPropsMixin {
...
+  /// Identifies the element that provides an error message for that object
+  ///
+  /// When there is a user-created error, you want to let them know it exists and tell them how to fix it. 
+  /// There are two attributes you need to use: set aria-invalid="true" to define the object as being in 
+  /// an error state, then add the aria-errormessage attribute with the value being the id of the element 
+  /// containing the error message text for that object
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage>
+  @Accessor(key: 'aria-errormessage')
+  String error;
```
Comment (greglittlefield-wf) [inline lib/src/component/aria_mixin.dart:269]: Is there a reason this is `error` instead of `errormessage`?
```suggestion
  String errormessage;
```

**Review by greglittlefield-wf — CHANGES_REQUESTED** [review]
Two tiny things otherwise this looks great.

Thanks for updating all the doc comments, too!

**Comment by sydneesampson-wk** [comment]
Looks like the build is failing on the analyzer test check, but this seems to be failing on the master branch as well. Please let me know how to go about this

**Comment by greglittlefield-wf** [comment]
@sydneesampson-wk We should be able to ignore that check for now, sorry about that!

File: lib/src/component/aria_mixin.dart [hunk 8]
```diff
@@ -34,13 +35,16 @@ abstract class $AriaPropsMixin {
   static const PropsMeta meta = _$metaForAriaPropsMixin;
 }
 
+///See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes>
+
```
Comment (aaronlademann-wf) [inline lib/src/component/aria_mixin.dart:39]: For this - when "documenting" an entire page / library, we usually put them above the library directive. In this case - that is on line 15...

```dart
/// See: <https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes>
library over_react.aria_mixin;
```

**Review by aaronlademann-wf — COMMENTED** [review]
Looks great! Just had one nitpick about a doc comment location

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**